### ISSUE TITLE
Can't mock duplicate services with CBMPeripheralSpecDelegate?

### ISSUE THREAD

**Issue by jaylyerly** [NordicSemiconductor/IOS-CoreBluetooth-Mock#42]
I'm interfacing with a device that has two battery level services.  (The device has two physical batteries.). In Core Bluetooth, I can distinguish these by resolving the descriptors on the characteristics.  The value of the `CBMUUIDCharacteristicFormatString` descriptor is different for the two services.  I think that's the correct way to do it according to the spec.

But, I run into a problem when I try to mock this object.  In `CBMPeripheralSpecDelegate`, I can implement the following method to return the data for the descriptor.
```swift
 func peripheral(_ peripheral: CBMPeripheralSpec, didReceiveReadRequestFor descriptor: CBMDescriptor) -> Result<Data, Error> 
```
But I don't think I have enough information here to distinguish between the two services.  The UUID for both the service and the characteristic are the standard 'battery level' values (180F / 2A19).  I don't see another way to differentiate between the two. 

Am I missing something?  Is there another identifier I can use?  Nothing stood out looking through the source code.  Is there another mechanism for supplying the `value` of a descriptor?

Thanks for the insight!

**Comment by philips77** [comment]
That's a good one. I didn't think of this use case. This is the only way to providing the value. Perhaps adding parent characteristic and service is a way to go.

**Comment by jaylyerly** [comment]
Yeah, if there were some way to distinguish them, that'd be great.

Another option might be to add a `userInfo` dictionary to the mock objects.  Kind of an old-school Cocoa pattern, but sometimes it's really useful just to be able to add some arbitrary info to the objects as they get passed around.

I've only been using the library for a couple days, but it's working really really well for us!  Thanks for creating it!

**Comment by philips77** [comment]
Hi,
Have a look at #62. I descried the changes there and added a test for a use case similar to your device.

**Comment by philips77** [comment]
Released in 0.13.0.

**Comment by philips77** [comment]
Let me know if something doesn't work as expected.
